src/frontends/pytorch/src/input_model.cpp

```diff
@@ -81,68 +75,117 @@ void InputModel::set_partial_shape(const Place::Ptr& place, const ov::PartialSha
-        FRONT_END_GENERAL_CHECK(el_type.is_static() && p_shape.is_static(),
-                                "Shape and type must be statically defined before calling set_tensor_value");
-        it->second.m_value = ov::op::v0::Constant::create(el_type, p_shape.to_shape(), value);
+        it->second.m_value = const_node;
     } else {
-        FRONT_END_GENERAL_CHECK(false, "Place is not known.");
+        m_descriptors.emplace(tensor_id, PlaceDesc(const_node));
     }
+    // remove place from inputs
+    m_inputs.erase(std::remove_if(m_inputs.begin(),
+                                  m_inputs.end(),
+                                  [&](const Place::Ptr& p) {
+                                      return p->is_equal(place);
+                                  }),
+                   m_inputs.end());
+}
+
+void InputModel::override_all_outputs(const std::vector<Place::Ptr>& outputs) {
+    // Topology modification is not allowed, but order can be changed
+    auto all_outputs = std::all_of(outputs.cbegin(), outputs.cend(), [](const Place::Ptr& p) {
+        return p->is_output();
+    });
+    FRONT_END_GENERAL_CHECK(all_outputs, "Only initial outputs are supported by override_all_inputs.");
+    // Number of outputs can be lower then in original model
+    m_outputs = outputs;
+}
+
+void InputModel::override_all_inputs(const std::vector<Place::Ptr>& inputs) {
+    // Topology modification is not allowed, but order can be changed
+    auto all_inputs = std::all_of(inputs.cbegin(), inputs.cend(), [](const Place::Ptr& p) {
+        return p->is_input();
+    });
+    FRONT_END_GENERAL_CHECK(all_inputs, "Only initial inputs are supported by override_all_inputs.");
+    // We need to add back "self" input if it was in initial inputs
+    if (m_inputs.size() > 0 && m_inputs[0]) {
+        // We need to remove "self" input to not confuse external users
+        const auto& names = m_inputs[0]->get_names();
+        if (std::any_of(names.cbegin(), names.cend(), [](const std::string& n) {
+                return n.find("self") != std::string::npos;
+            })) {
+            FRONT_END_GENERAL_CHECK(inputs.size() == m_inputs.size() - 1,
+                                    "Number of inputs provided is incorrect. Graph modification is not supported for "
+                                    "this model. Expected number of inputs: ",
+                                    m_inputs.size() - 1,
+                                    " recieved ",
+                                    inputs.size());
+            auto self_place = m_inputs[0];
+            // Verify that no same place already in vector
+            auto no_self = std::all_of(inputs.cbegin(), inputs.cend(), [&](const Place::Ptr& p) {
+                return !p->is_equal(self_place);
+            });
+            FRONT_END_GENERAL_CHECK(no_self,
+                                    "Unexpected input place of 'self' input was provided to override_all_inputs.");
```

Comment:
Should this be "Unexpected input place of 'self'. Input was provided to override_all_inputs."?

Reply:
No, but agree, this is complicated sentence. The subject here is "input place" that represent "self input", I will simplify it.